scripts/firedrake-configure

```diff
@@ -0,0 +1,436 @@
+        if package_manager is None:
+            # Don't know anything about the system, download everything
+            package_spec = PETSC_DOWNLOAD
+        else:
+            package_spec = PETSC_EXTERNAL_PACKAGE_SPECS[external_package][package_manager]
+
+        if package_spec == PETSC_AUTODETECT:
+            # PETSc will find the package for us
+            configure_options.append(f"--with-{external_package}")
+        elif package_spec == PETSC_DOWNLOAD:
+            # Package not provided by package manager, download it instead
+            configure_options.append(f"--download-{external_package}")
+        elif isinstance(package_spec, str):
+            # Package is installed in a 'sensible' way to <prefix>/include and
+            # <prefix>/lib but PETSc does not know <prefix> so we have to provide it
+            configure_options.append(f"--with-{external_package}-dir={package_spec}")
+        else:
+            # Package is installed but not findable, have to provide paths
+            # to headers and libraries
+            (include_dir, libs) = package_spec
+            configure_options.append(f"--with-{external_package}-include={include_dir}")
+            configure_options.append(f"--with-{external_package}-lib={','.join(libs)}")
+    return tuple(configure_options)
+
+
+def prepare_configure_options(
+    package_manager: Optional[PackageManager],
+    arch: FiredrakeArch,
+) -> tuple[str, ...]:
+    configure_options = list(COMMON_PETSC_CONFIGURE_OPTIONS)
+    configure_options.append(f"PETSC_ARCH=arch-firedrake-{arch.value}")
+
+    if arch == ARCH_COMPLEX:
+        configure_options.append("--with-scalar-type=complex")
+
+    external_packages = list(COMMON_PETSC_EXTERNAL_PACKAGES)
+    if arch != ARCH_COMPLEX:
+        external_packages.append("hypre")
+    configure_options.extend(
+        prepare_external_package_configure_options(external_packages, package_manager)
+    )
+    return tuple(configure_options)
+
+
+PETSC_CONFIGURE_OPTIONS = {
+    (package_manager, arch): prepare_configure_options(package_manager, arch)
+    for (package_manager, arch) in (
+        (LINUX_APT_X86_64, ARCH_DEFAULT),
+        (LINUX_APT_X86_64, ARCH_COMPLEX),
+        (LINUX_APT_AARCH64, ARCH_DEFAULT),
+        (LINUX_APT_AARCH64, ARCH_COMPLEX),
+        (MACOS_HOMEBREW_ARM64, ARCH_DEFAULT),
+        (MACOS_HOMEBREW_ARM64, ARCH_COMPLEX),
+        (None, ARCH_DEFAULT),
+        (None, ARCH_COMPLEX),
+    )
+}
+
+
+def prepare_environment_vars(
+    package_manager: Optional[PackageManager],
+    arch: FiredrakeArch,
+) -> tuple[str, ...]:
+    vars = {
+        "CC": "mpicc",
+        "CXX": "mpicxx",
+        "PETSC_DIR": f"{os.getcwd()}/petsc",
+        "PETSC_ARCH": f"arch-firedrake-{arch.value}",
+        "HDF5_MPI": "ON",
+    }
+
+    # On macOS h5py cannot find the HDF5 library without help
+    if package_manager == MACOS_HOMEBREW_ARM64:
+        vars["HDF5_DIR"] = "/opt/homebrew"
```

Comment:
```suggestion
    if package_manager == MACOS_HOMEBREW_ARM64:
        vars["HDF5_DIR"] = "/opt/homebrew"
    elif package_manager is None:
        vars["HDF5_DIR"] = vars["PETSC_DIR"]+"/"+vars["PETSC_ARCH"]
```
If `--no-package-manager` is used then PETSc will be told to download hdf5, and `HDF5_DIR` must be set so `h5py` knows where to link against.

Reply:
Good spot. Done.